src/Symfony/Component/Notifier/Bridge/Iterable/IterableTransport.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Notifier\Bridge\Iterable;
+
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Balázs Csaba <csaba.balazs@lingoda.com>
+ */
+final class IterableTransport extends AbstractTransport
+{
+    protected const HOST = 'api.iterable.com/api/push/target';
+
+    /** @var string */
+    private $apiKey;
+
+    /** @var string|null */
+    private $campaignId;
+
+    public function __construct(string $apiKey, ?string $campaignId, ?HttpClientInterface $client = null, ?EventDispatcherInterface $dispatcher = null)
+    {
+        $this->apiKey = $apiKey;
+        $this->campaignId = $campaignId;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        if (null !== $this->campaignId) {
+            return sprintf('iterable://%s?campaign_id=%s', $this->getEndpoint(), $this->campaignId);
+        }
+
+        return sprintf('iterable://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        if ($message->getOptions() && !$message->getOptions() instanceof IterableOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, IterableOptions::class));
+        }
+
+        $endpoint = sprintf('https://%s', $this->getEndpoint());
+        $options = ($opts = $message->getOptions()) ? $opts->toArray() : [];
+
+        $campaignId = $options['campaignId'] ?? $this->campaignId;
+        if (null === $campaignId) {
+            throw new InvalidArgumentException(sprintf('The "%s" transport required the "campaignId" to be set.', __CLASS__));
```

Comment:
Is it really a common use case to not set a campaign_id on the DSN and set different ones on the ChatMessage via options?

If not, I would make the campaign_id required and not nullable and remove it from the IterableOptions